spharpy/classes/ambisonics.py

```diff
@@ -0,0 +1,217 @@
+            converted to ``complex`` and must be provided as single
+            sided spectra, i.e., for all frequencies between 0 Hz and
+            half the sampling rate.
+        sampling_rate : double
+            Sampling rate in Hz
+        n_max : int
+            Maximum spherical harmonic order. Has to match the number of
+            coefficients, such that the number of coefficients
+            .. math:: >= (n_max + 1) ** 2.
+        basis_type : str
+            Type of spherical harmonic basis, either ``'complex'`` or
+            ``'real'``. The default is ``'real'``.
+        normalization : str
+            Normalization convention, either ``'n3d'``, ``'maxN'`` or
+            ``'sn3d'``. The default is ``'n3d'``.
+            (maxN is only supported up to 3rd order)
+        channel_convention : str
+            Channel ordering convention, either ``'acn'`` or ``'fuma'``.
+            The default is ``'acn'``.
+            (FuMa is only supported up to 3rd order)
+        n_samples : int, optional
+            Number of samples of the time signal. Required if domain is
+            ``'freq'``. The default is ``None``, which assumes an even number
+            of samples if the data is provided in the frequency domain.
+        domain : ``'time'``, ``'freq'``, optional
+            Domain of data. The default is ``'time'``
+        fft_norm : str, optional
+            The normalization of the Discrete Fourier Transform (DFT). Can be
+            ``'none'``, ``'unitary'``, ``'amplitude'``, ``'rms'``, ``'power'``,
+            or ``'psd'``. See :py:func:`~pyfar.dsp.fft.normalization` and [#]_
+            for more information. The default is ``'none'``, which is typically
+            used for energy signals, such as impulse responses.
+        comment : str
+            A comment related to `data`. The default is ``None``.
+
+        References
+        ----------
+        .. [#] E.G. Williams, "Fourier Acoustics", (1999), Academic Press
+        .. [#] B. Rafely, "Fundamentals of Spherical Array Processing", (2015),
+               Springer-Verlag
+        .. [#] F. Zotter, M. Frank, "Ambisonics A Practical 3D Audio Theory
+               for Recording, Studio Production, Sound Reinforcement, and
+               Virtual Reality", (2019), Springer-Verlag
+
+        """
+    def __init__(
+            self,
+            data,
+            sampling_rate,
+            n_max,
+            basis_type,
+            normalization,
+            channel_convention,
+            phase_convention=None,
+            n_samples=None,
+            domain='time',
+            fft_norm='none',
+            comment="",
+            is_complex=False):
+        """
+        Create SphericalHarmonicSignal with data, and sampling rate.
+        """
+        # set n_max
+        if n_max < 0:
+            raise ValueError("n_max must be a positive integer")
+        if n_max % 1 != 0:
+            raise ValueError("n_max must be an integer value")
```

Comment:
```suggestion
        if (n_max < 0) or (n_max % 1 != 0):
            raise ValueError("n_max must be a positive integer")
```
maybe we can combine it. Alternativly we could check isinstance(n_max, int), so we force int, as mentioned in the doc